# SPDX-License-Identifier: MIT OR BSL-1.0
# Copyright (c) 2026 openEuler RISC-V packaging contributors.
# Subject: Repair original test-driver wiring and release metadata for 0.6.3
# Upstream: https://github.com/docopt/docopt.cpp
# Applies-to: v0.6.3 / commit 42ebcec9dc2c99a1b3a4542787572045763ad196
# Source-archive-SHA256: 334f5b9d1cb7d437c62282f0a97370ddfdfbe33de02428be9b79c7719a8a57f3
# Status: Packaging proposal only; not submitted upstream.
# Root cause: WITH_TESTS registers the case-runner without its required USAGE
# argument and never enables CTest, while upstream CI runs the configured
# Python 2 driver. Use Python 3 print syntax and register that same full driver.
# The formal 0.6.3 release retains a stale 0.6.2 CMake version; correct only the
# release stamp, preserving SONAME major 0 and original shared/static targets.
# Scope: generic build/test metadata, not RISC-V-only; no parser/case changes.
# Preserve: WITH_TESTS/WITH_EXAMPLE/USE_BOOST_REGEX defaults OFF, C++11,
# all original 175 data cases and expected JSON/error outcomes unchanged.
# Remove when an accepted official stable source supplies equivalent fixes.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2 +2 @@
-project(docopt.cpp VERSION 0.6.2)
+project(docopt.cpp VERSION 0.6.3)
@@ -101,0 +102,2 @@
+	find_package(Python3 3.6 REQUIRED COMPONENTS Interpreter)
+	enable_testing()
@@ -111 +113,2 @@
-	add_test("Testcases docopt" ${TESTPROG})
+	add_test(NAME docopt-testcases
+		COMMAND "${Python3_EXECUTABLE}" "${CMAKE_CURRENT_BINARY_DIR}/run_tests")
--- a/run_tests.py
+++ b/run_tests.py
@@ -1 +1 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
@@ -59,5 +59,5 @@
-		print "="*40
-		print doc
-		print ':'*20
-		print prog, argv
-		print '-'*20
+		print("="*40)
+		print(doc)
+		print(':'*20)
+		print(prog, argv)
+		print('-'*20)
@@ -65,2 +65,2 @@
-			print out
-		print error
+			print(out)
+		print(error)
@@ -69 +69 @@
-	print "%d failures" % failures
+	print("%d failures" % failures)
@@ -72 +72 @@
-	print "PASS (%d)" % passes
+	print("PASS (%d)" % passes)
